strapi/strapi
# Allow adding models from extensions

### What does it do?

When loading overwrites for plugins, if a model found in the overwrites (./extensions/ folder) does not exist on a plugins object, we add the value of that model at the path specified in the overwrites.

### Why is it needed?

Use case:
When extending a users-permissions plugin need to create a new model. Adding new model to the extensions folder does not work without this change. The only option is to fork the Strapi and make changes directly in the source of the users-permissions plugin.

With this change, the models will be added to the plugin upon strapi startup.

### How to test it?

Extend users-permissions plugin and add a new model in the extension. The model should be available later to use in code.

### Related issue(s)/PR(s)
Nothing really except my question in the forum that is not answered so far: https://forum.strapi.io/t/add-new-model-to-strapi-plugin-users-permissions/9242
